optuna_dashboard/preferential/_study.py

```diff
@@ -269,15 +243,16 @@ def should_generate(self) -> bool:
 def get_best_trials(study_id: int, storage: optuna.storages.BaseStorage) -> list[FrozenTrial]:
     preferences = get_preferences(study_id, storage)
     worse_numbers = {worse for _, worse in preferences}
+    nondominated_numbers = {better for better, _ in preferences if better not in worse_numbers}
+    trials = storage.get_all_trials(
+        study_id, deepcopy=False, states=(TrialState.COMPLETE, TrialState.RUNNING)
+    )
+
     study_system_attrs = storage.get_study_system_attrs(study_id)
+
     best_trials = []
-    for t in storage.get_all_trials(
-        study_id, deepcopy=False, states=(TrialState.COMPLETE, TrialState.RUNNING)
-    ):
-        if not t.system_attrs.get(_SYSTEM_ATTR_COMPARISON_READY, False):
-            continue
-        if t.number in worse_numbers:
-            continue
+    for n in nondominated_numbers:
+        t = trials[n]
```

Comment:
We cannot access `trials[number]` when filtering trials by the trial state. Could you fix it?

Reply:
Thank you for pointing that out!

Reply:
I fixed it.